### ISSUE TITLE
Idea: Emit events as the finality checker performs its checks

### ISSUE THREAD

**Issue by aaroncox** [wharfkit/transact-plugin-finality-checker#6]
Got a recommendation today that it'd be nice if this finality "monitor" (or whatever we're calling this one) could somehow emit events to the frontend related to finality, so that the app could then handle those results in their own UI. 

I don't know the structure of how we relay these events back to the app, but I imagine every time we call get_transaction_status we can relay that info back to the app. Then when the finality threshold is reached, we call the callback like it does now.

**Comment by dafuga** [comment]
Yeah, that would be cool. Maybe the FE can pass an `updateCallback` function as a parameter when initializing the package?

**Comment by dafuga** [comment]
And maybe that should be a parameter in the transact-plugin-finality-callback plugin instead of this one?
